core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/AbstractSimplePropertyShape.java

```diff
@@ -101,7 +101,7 @@ static public PlanNode getGenericSingleObjectPlan(ConnectionsGroup connectionsGr
 
 			PlanNode discardedRight = innerJoin.getDiscardedRight(BufferedPlanNode.class);
 
-			PlanNode typeFilterPlan = nodeShape.getTargetFilter(connectionsGroup.getBaseConnection(), discardedRight);
+			PlanNode typeFilterPlan = nodeShape.getTargetFilter(connectionsGroup, discardedRight);
```

Comment:
This is to be able to make sure that the getTargetFilter on NodeShape only operates on the base connection.